From 35eab81126aa8cf8e39ff4cc3da1fd087c13ab67 Mon Sep 17 00:00:00 2001
From: Florent Revest <revestflo@gmail.com>
Date: Fri, 19 Jul 2019 19:31:27 +0200
Subject: [PATCH 2/3] hci: don't set MWS command

The HCI_OP_GET_MWS_TRANSPORT_CONFIG command isn't properly handled by
the BT chip and causes our BT initialization to fail. To workaround
this we're not sending this command for now as MWS isn't used anywhere
yet.
---
 net/bluetooth/hci_core.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/net/bluetooth/hci_core.c b/net/bluetooth/hci_core.c
index 62edbf1b114e..85bca498d508 100644
--- a/net/bluetooth/hci_core.c
+++ b/net/bluetooth/hci_core.c
@@ -893,9 +893,16 @@ static void hci_init4_req(struct hci_request *req, unsigned long opt)
 	if (hdev->commands[29] & 0x20)
 		hci_req_add(req, HCI_OP_READ_LOCAL_CODECS, 0, NULL);
 
+	/* NOTE: We're not sending the HCI_OP_GET_MWS_TRANSPORT_CONFIG
+	 * command here as it will come back with an error and will
+	 * cause the whole initialization to fail. This is a misbehaviour of
+	 * the BT chip but can't be fixed. Dropping this will not cause
+	 * any side effects as nobody is using MWS yet. */
+#if 0
 	/* Get MWS transport configuration if the HCI command is supported */
 	if (hdev->commands[30] & 0x08)
 		hci_req_add(req, HCI_OP_GET_MWS_TRANSPORT_CONFIG, 0, NULL);
+#endif
 
 	/* Check for Synchronization Train support */
 	if (lmp_sync_train_capable(hdev))
-- 
2.22.0

